sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1201 @@
+            which only applies to the stored `self`. If no `self` routing
+            object is stored, this parameter has no effect.
+
+        ignore_self : bool
+            If `self._self` should be ignored. This is used in
+            `_get_squashed_params`. If ``True``, ``original_names`` has no
+            effect.
+
+        Returns
+        -------
+        names : set of str
+            Returns a set of strings with the names of all parameters.
+        """
+        res = set()
+        if self._self and not ignore_self:
+            res = res.union(
+                self._self._get_param_names(
+                    method=method, original_names=original_names
+                )
+            )
+
+        for name, route_mapping in self._route_mappings.items():
+            for orig_method, used_in in route_mapping.mapping:
+                if used_in == method:
+                    res = res.union(
+                        route_mapping.routing._get_param_names(
+                            method=orig_method, original_names=False, ignore_self=False
+                        )
+                    )
+        return set(sorted(res))
+
+    def _get_squashed_params(self, *, params, method):
+        """Get input for a method of a router w/o validation.
+
+        This is used when a router is used as a child object of another router.
+        The parent router then passes all parameters understood by the child
+        object to it and delegates their validation to the child.
+
+        The output of this method can be used directly as the input to the
+        corresponding method as extra props.
+
+        Parameters
+        ----------
+        method : str
+            The name of the method for which the parameters are requested and
+            routed.
+
+        params : dict
+            A dictionary of provided metadata.
+
+        Returns
+        -------
+        params : Bunch
+            A :class:`~utils.Bunch` of {prop: value} which can be given to the
+            corresponding method.
+        """
+        res = Bunch()
+        if self._self:
+            res.update(self._self._get_params(params=params, method=method))
+        param_names = self._get_param_names(
+            method=method, original_names=False, ignore_self=True
+        )
+        res.update({key: value for key, value in params.items() if key in param_names})
+        return res
+
+    def get_params(self, *, method, params):
+        """Return the input parameters requested by a child objects.
+
+        The output of this method is a bunch, which includes the inputs for all
+        methods of each child object that are used in the router's `"method"`.
+
+        If the router is also a consumer, it also checks for warnings of
+        `self`'s/consumer's requested metadata.
+
+        Parameters
+        ----------
+        method : str
+            The name of the method for which the parameters are requested and
```

Comment:
It's unclear if this is the router method or the destination method name. from the above context, I think you mean the router method. Can we rename this `router_method`? Will that be clearer at all?

Reply:
Added an extra sentence to docs, and renamed the method, should be clear, rather not rename it since it makes it slightly too verbose.

Reply:
I think this name should be consistent with the name we use for `Route`, which is currently "used_in". If we go with "src", then "src_method".

I'm okay with a more verbose name if it makes the method clearer.